File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/554-brick-wall.js
var leastBricks = function(wall) {
    const edgeCount = new Map();
    let maxEdges = 0;

    for (const row of wall) {
        let edgePos = 0;
        for (let i = 0; i < row.length - 1; i++) {
            edgePos += row[i];
            edgeCount.set(edgePos, (edgeCount.get(edgePos) || 0) + 1);
            maxEdges = Math.max(maxEdges, edgeCount.get(edgePos));
        }
    }

    return wall.length - maxEdges;
};

File: leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/554-brick-wall.erl
-spec least_bricks(Wall :: [[integer()]]) -> integer().
least_bricks(Wall) ->
    EdgeCount = maps:new(),
    {EdgeCount1, MaxEdges} = lists:foldl(
        fun(Row, {Count, Max}) ->
            {NewCount, NewMax} = lists:foldl(
                fun(Brick, {AccCount, AccMax}) ->
                    case maps:get(Brick, AccCount, 0) of
                        undefined -> {maps:put(Brick, 1, AccCount), max(AccMax, 1)};
                        N -> {maps:put(Brick, N + 1, AccCount), max(AccMax, N + 1)}
                    end
                end,
                {Count, Max},
                lists:sublist(Row, 1, length(Row) - 1)
            ),
            {NewCount, NewMax}
        end,
        {EdgeCount, 0},
        Wall
    ),
    length(Wall) - MaxEdges.